### PR TITLE
Bootstrap classification template repo

### PR DESCRIPTION
**Pull request by bbarnes52**
This is a copy of the regression template repo with the following changes:
- Builtin metrics are updated to classifier metrics
- All instances of `regression` and its variants are updated to `classification` and its corresponding variants
- Added a note about requirement that `target_col` must have a cardinality of two

### CODE REVIEW

File: recipe.yaml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+#       to adapt this template to a specific classification problem. To find all required fields,
+#       under the root directory of this recipe, type on a unix-like command line:
+#       $> grep "# FIXME::REQUIRED:" recipe.yaml profiles/*.yaml
+#
+# NOTE: YAML does not support tabs for indentation. Please use spaces and ensure that all YAML
+#       files are properly formatted.
+
+template: "classification/v1"
+# FIXME::REQUIRED: Specifies the target column name for model training and evaluation.
+target_col: ""
```
Comment (prithvikannan) [inline recipe.yaml:20]: Let's add the `positive_class` field here? And mark it as `FIXME::REQUIRED`

**Review by prithvikannan — COMMENTED** [review]
Nice! Left a note about including the `positive_class` field.

File: notebooks/databricks.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+# MAGIC # MLflow Classification Recipe Databricks Notebook
+# MAGIC This notebook runs the MLflow Classification Recipe on Databricks and inspects its results.
+# MAGIC
+# MAGIC For more information about the MLflow Classification Recipe, including usage examples,
+# MAGIC see the [Classification Recipe overview documentation](https://mlflow.org/docs/latest/recipes.html#classification-recipe)
+# MAGIC and the [Classification Recipe API documentation](https://mlflow.org/docs/latest/python_api/mlflow.recipes.html#module-mlflow.recipes.classification.v1.recipe).
+
+# COMMAND ----------
+
+# MAGIC %pip install mlflow[recipes]
```
Comment (sunishsheth2009) [inline notebooks/databricks.py:13]: I don't think we have this anymore. 

We just need `pip install mlflow`

File: requirements.txt [hunk 3]
```diff
@@ -0,0 +1,6 @@
+mlflow[pipelines]>=1.27.0
```
Comment (sunishsheth2009) [inline requirements.txt:1]: Same here. We can this to `mlflow>=2.0.0rc0`

File: recipe.yaml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+#
+# NOTE: All "FIXME::REQUIRED" fields in recipe.yaml and profiles/*.yaml must be set correctly
+#       to adapt this template to a specific classification problem. To find all required fields,
+#       under the root directory of this recipe, type on a unix-like command line:
+#       $> grep "# FIXME::REQUIRED:" recipe.yaml profiles/*.yaml
+#
+# NOTE: YAML does not support tabs for indentation. Please use spaces and ensure that all YAML
+#       files are properly formatted.
+
+template: "classification/v1"
```
Comment (sunishsheth2009) [inline recipe.yaml:18]: Xiangrui Meng suggested to change this to `recipe: "classification/v1"`. I am going to make the corresponding change in mlflow, example repo and regression-template repo now.

**Review by prithvikannan — APPROVED** [review]
LGTM! Let's wait for https://github.com/mlflow/mlflow/pull/7244 to merge.

File: steps/custom_metrics.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    :param builtin_metrics: A dictionary containing the built-in metrics that are calculated
+                            automatically during model evaluation. The keys are the names of the
+                            metrics and the values are the scalar values of the metrics. For more
+                            information, see
+                            https://mlflow.org/docs/latest/python_api/mlflow.html#mlflow.evaluate.
+    :return: A single-entry dictionary containing the custom metrics. The key is the metric name
+             and the value is the scalar metric value. Note that custom metric functions can
+             return dictionaries with multiple metric entries as well.
+    """
+    # FIXME::OPTIONAL: implement custom metrics calculation here.
```
Comment (sunishsheth2009) [inline steps/custom_metrics.py:38]: This is fixed by @harupy in the new [regression template](https://github.com/mlflow/recipes-regression-template/blob/main/steps/custom_metrics.py#L22). Can we replicate that here?

File: requirements.txt [hunk 6]
```diff
@@ -0,0 +1,6 @@
+mlflow>=2.0.0rc0
+pandas>=1.3.*
+scikit-learn>=1.1.*
+ipykernel>=6.12.*
+ipython>=7.32.*
+shap>=0.40.*
```
Comment (sunishsheth2009) [inline requirements.txt:6]: I think pandas, sklearn and shap are no more needed: https://github.com/mlflow/recipes-regression-template/blob/main/requirements.txt

They are included in mlflow directly.

**Pull request merged**